clang/include/clang/Basic/FileEntry.h

```diff
@@ -70,6 +70,13 @@ class FileEntryRef {
   const FileEntry &getFileEntry() const {
     return *getBaseMapEntry().second->V.get<FileEntry *>();
   }
+
+  // This is a non const version of getFileEntry() which is used if the buffer
+  // size needs to be increased due to potential z/OS EBCDIC -> UTF-8 conversion
+  FileEntry &getFileEntryToUpdate() {
```

Comment:
nit: Consider rename this to `updateFileEntry()`

Reply:
I think this would not be an accurate name because the function itself does not do any updates, it just returns a non const FileEntry. I also did not want to keep the same name as getFileEntry because that is const function. I'm open to alternative names, maybe `getFileEntryForUpdate`?

Reply:
We could convert this function to do the update by taking the `BufferSize` as parameter, isn't it?

Reply:
Thanks that's a good idea, I had to move the function definition below the class FileEntry and have implemented your suggested solution now